src/dali/abstract_pair_converter_plugin.py

```diff
@@ -179,35 +198,32 @@ def _build_fiat_list(self) -> None:
             LOGGER.info("Fetching of fiat symbols failed. The server might be down. Please try again later.")
             raise RP2RuntimeError("JSON decode error") from exc
 
-    def _add_fiat_edges_to_graph(self, graph: Dict[str, Dict[str, None]], markets: Dict[str, List[str]]) -> None:
+    def _add_fiat_edges_to_graph(self, graph: Graph[str], markets: Dict[str, List[str]]) -> None:
+        vertexes: Dict[str, Vertex[str]] = {v.name: v for v in graph.vertexes}
```

Comment:
Regarding your question on vertexes, I would avoid adding the mapping to the Graph class in prezzemolo because we're trying to keep it small: this mapping can be easily managed in the application (which is adding the vertexes after all). So I would have a class dictionary here that manages name -> vertex (and some protected accessors to make it usable in subclasses). Basically every time you add a new vertex to the graph you would also update the dictionary.